src/dcap_provider.cpp

```diff
@@ -300,7 +328,7 @@ static std::string build_pck_cert_url(const sgx_ql_pck_cert_id_t& pck_cert_id)
         format_as_big_endian_hex_string(pck_cert_id.pce_id);
 
     return get_base_url() + '/' + qe_id + '/' + cpu_svn + '/' + pce_svn + '/' +
-           pce_id + '?' + API_VERSION;
+           pce_id + '?' + "clientid=" + get_client_id() + '&' + API_VERSION;
 }

```

Comment:
It looks like the change make "clientid=" a required parameter for PCK Caching service, right?
That is, it needs to treat pce_svn/pce_id/clientid=/API_VERSION a valid request string.

Reply:
Yes this is a good point. Is this field going to be required in the future? Maybe, if the environment variable isn't set, we shouldn't even reference the "clientid" field in the URL?  What do you think would be best @msft-jonnadul?

Reply:
I confirmed that append a undefined clientid value to the calls into the PCK cache end point don't seem to regress previous behavior, ex: this query will resolve currently https://global.acccache.azure.net/sgx/certificates/00bd4b2879d5a2764a964ab990908b67/05050205ff8001000000000000000000/0600/0000?clientid=&api-version=2018-10-01-preview.

Reply:
But I'll modify it such that if a clientid isn't defined, then it'll not even attempt to populate this variable.